zkvm/tests/zkvm.rs

```diff
@@ -0,0 +1,72 @@
+use bulletproofs::r1cs;
+use bulletproofs::{BulletproofGens, PedersenGens};
+use curve25519_dalek::ristretto::CompressedRistretto;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use std::collections::VecDeque;
+
+use zkvm::*;
+
+fn issue_contract(qty: u64, flv: Scalar, pred: &Predicate) -> Vec<Instruction> {
+    vec![
+        // pushdata qty [qty]
+        Instruction::Push(Commitment::from(CommitmentWitness::from_secret(qty)).into()),
+        // stack: qty-var
+        Instruction::Var,
+        // stack: flv, qty-var
+        Instruction::Push(Commitment::from(CommitmentWitness::unblinded(flv)).into()),
+        // stack: flv-var, qty-var
+        Instruction::Var,
+        // stack: pred, flv-var, qty-var
+        Instruction::Push(pred.clone().into()),
+        // stack: issue-contract
+        Instruction::Issue,
+        // stack: pred, issue-contract
+        Instruction::Push(pred.clone().into()),
```

Comment:
reusing the predicate would work for the unit test, but it's not going to be workable in real life: the nonces must be unique and by default would use random predicate keys. I'd keep that explicit in the tests to avoid confusion later. We can pass `issuance_pred` and `nonce_pred` as arguments to this method.